### PR TITLE
ENH: Do not require file type in sample data specification

### PR DESCRIPTION
**Pull request by lassoan**
Previously, `VolumeFile` type was used by default if type was not explicitly specified for a sample data item. Since readers can now provide confidence value, the default type is most often correct and it is no longer necessary to specify file type.

### CODE REVIEW

File: Modules/Scripted/SampleData/SampleData.py [hunk 1]
```diff
@@ -157,16 +158,22 @@ def __init__(self, sampleName=None, sampleDescription=None, uris=None, fileNames
...
         :param customDownloader: Custom function for downloading.
-        :param loadFileType: file format name(s) ('VolumeFile' by default if node name is specified).
+        :param loadFileTypes: file format name(s) (if not specified then the default file reader will be used).
         :param loadFileProperties: custom properties passed to the IO plugin.
+        :param loadFileType: deprecated, use ``loadFileTypes`` instead.
         """
+
+        # For backward compatibility (allow using "loadFileType" instead of "loadFileTypes")
+        if (loadFileType is not None) and (loadFileTypes is None):
+            loadFileTypes = loadFileType
```
Comment (jcfr) [inline Modules/Scripted/SampleData/SampleData.py:168]: I suggest to raise an exception[^1] if both `loadFileType` and `loadFileTypes` are specified. Updated code should only use `loadFileTypes`

[^1]: I suggest `ValueError`. See https://stackoverflow.com/questions/256222/which-exception-should-i-raise-on-bad-illegal-argument-combinations-in-python

File: Modules/Scripted/SampleData/SampleData.py [hunk 2]
```diff
@@ -178,24 +185,9 @@ def __init__(self, sampleName=None, sampleDescription=None, uris=None, fileNames
...
+            loadFileTypes = [loadFileTypes]
             checksums = [checksums]
 
-        updatedFileType = []
-        for fileName, nodeName, fileType in zip(fileNames, nodeNames, loadFileType):
-            # If not explicitly specified, attempt to guess fileType
-            if fileType is None:
-                if nodeName is not None:
-                    # TODO: Use method from Slicer IO logic ?
-                    fileType = "VolumeFile"
```
Comment (jcfr) [inline Modules/Scripted/SampleData/SampleData.py:190]: What about keeping this logic and instead have something like
```
                    fileType = None  # the fileType will be set automatically based on the registered readers.
```

File: Modules/Scripted/SampleData/SampleData.py [hunk 3]
```diff
@@ -684,6 +692,14 @@ def downloadFromSource(self, source, maximumAttemptsCount=3):
                     continue
                 resultFilePaths.append(filePath)
 
+                # Special case for backward compatibility for loading scene and zip files when loadFileType is not specified
```
Comment (jcfr) [inline Modules/Scripted/SampleData/SampleData.py:695]: I suggest to keep the logic  in the `SampleDataSource` and update the code as described in https://github.com/Slicer/Slicer/pull/7479/files#r1428946872

Comment (jcfr) [reply Modules/Scripted/SampleData/SampleData.py:695]: Does the comment mean that if the filetype is explicitly set to `SceneFile` or `ZipFile`, the behavior will be different ? 

I would suggest to have the same behavior .. may be the comment should instead be:

```

                # Special behavior to handle "ZipFile" and "SceneFile"
```

File: Modules/Scripted/SampleData/SampleData.py [hunk 4]
```diff
@@ -916,11 +937,15 @@ def loadScene(self, uri, fileProperties={}):
...
         return True
 
-    def loadNode(self, uri, name, fileType="VolumeFile", fileProperties={}):
+    def loadNode(self, uri, name, fileType=None, fileProperties={}):
         self.logMessage("<b>" + _("Requesting load {name} from {uri} ...").format(name=name, uri=uri) + "</b>")
 
         fileProperties["fileName"] = uri
         fileProperties["name"] = name
+        if not fileType:
+            print(f"loadNode: not fileType, uri = {uri}")
```
Comment (jcfr) [inline Modules/Scripted/SampleData/SampleData.py:946]:
```suggestion
```

File: Modules/Scripted/SampleData/SampleData.py [hunk 5]
```diff
@@ -916,11 +937,15 @@ def loadScene(self, uri, fileProperties={}):
...
+            fileType = slicer.app.coreIOManager().fileType(fileProperties["fileName"])
+            print(f"loadNode: fileType = {fileType}")
```
Comment (jcfr) [inline Modules/Scripted/SampleData/SampleData.py:948]:
```suggestion
```

File: Modules/Scripted/SampleData/SampleData.py [hunk 6]
```diff
@@ -232,14 +224,16 @@ def __str__(self):
...
+            self.fileNames, self.uris, self.nodeNames, self.loadFiles, self.loadFileTypes, self.checksums):
+
             output.extend([
-                " fileName    : %s" % fileName,
-                " uri         : %s" % uri,
-                " checksum    : %s" % checksum,
-                " nodeName    : %s" % nodeName,
-                " loadFile    : %s" % loadFile,
-                " loadFileType: %s" % fileType,
+                " fileName     : %s" % fileName,
```
Comment (jcfr) [inline Modules/Scripted/SampleData/SampleData.py:231]: I suggest to add a first commit titled `STYLE: Improve SampleData source code readability` where all stylistic changes would be done, we will then add to it the `.git-blame-ignore-revs`[^1] file once the MR is merged.

[^1]: https://github.com/Slicer/Slicer/blob/main/.git-blame-ignore-revs

File: Modules/Scripted/SampleData/SampleData.py (hunk 2 above)
Comment (lassoan) [reply Modules/Scripted/SampleData/SampleData.py:190]: The behavior has not changed, the logic just moved to lower level.

File: Modules/Scripted/SampleData/SampleData.py (hunk 5 above)
Comment (lassoan) [reply Modules/Scripted/SampleData/SampleData.py:948]: Thank you, good catch.

File: Modules/Scripted/SampleData/SampleData.py (hunk 4 above)
Comment (lassoan) [reply Modules/Scripted/SampleData/SampleData.py:946]: Thank you, good catch.

**Comment by lassoan** [comment]
@jcfr Thanks for the review and suggestions. I've responded to all comments, please have a look.

**Pull request merged**